python/ml4ir/base/model/scoring/interaction_model.py

```diff
@@ -35,9 +35,13 @@ def __call__(self, inputs: Dict[str, Input]):
             computing custom metrics and losses
         """
         train_features, metadata_features = self.feature_layer_op(inputs)
-        train_features, metadata_features = self.transform_features_op(
-            train_features, metadata_features
-        )
+        # separate the default learnable features from the fr_features
+        if self.feature_config.feature_exists('fr', trainable=True):
+            train_fr_features, _ = self.transform_features_op({'fr_feature':train_features['fr_feature']}, metadata_features)
+            train_features, metadata_features = self.transform_features_op({k: train_features[k] for k in train_features if k != 'fr_feature'}, metadata_features)
+            return [train_features, train_fr_features], metadata_features
```

Comment:
@mohazahran Definitely get rid of this. Why are you returning these two? More importantly, what are these two? Keep in mind you are also introducing inconsistent number of return values within this method. Overall, I don't think we need this `fr_feature` and we could compute it on the fly within the positional bias layer itself.

Reply:
I'm returning two because I'm separating the normal features (sim score, pv ...) from the positional bias features (fr)